src/LiveComponent/src/Test/TestLiveComponent.php

```diff
@@ -89,7 +90,20 @@ public function call(string $action, array $arguments = []): self
      */
     public function emit(string $event, array $arguments = []): self
     {
-        return $this->call($event, $arguments);
+        $listeners = AsLiveComponent::liveListeners($this->component());
+        $actions = [];
+
+        foreach ($listeners as $listener) {
+            if ($listener['event'] === $event) {
+                $actions[] = ['name' => $listener['action'], 'args' => $arguments];
+            }
+        }
+
+        if (!$actions) {
+            throw new \InvalidArgumentException(sprintf('Event "%s" does not exist on component "%s".', $event, $this->metadata->getName()));
+        }
+
+        return $this->request(['actions' => $actions], '_batch');
```

Comment:
Super minor, and it shouldn't make any difference, but if we have just one action, it won't process through batch: it'll just call that one action directly.

Reply:
The js makes this distinction?

Reply:
Exactly. The js always tries to use the normal endpoint (either for calling a `LiveAction` or the "default" action in the case of a re-render). If it finds that it has 2 things queued at once, then it goes through batch. https://github.com/symfony/ux/blob/2.x/src/LiveComponent/assets/src/Backend/RequestBuilder.ts#L71-L79

Reply:
Ok, I'll match in the helper.

Reply:
This change has been made.